src/main/service/listManipulation/magsStandardListService.ts

```diff
@@ -0,0 +1,92 @@
+import moment from 'moment-timezone';
+import {DataManipulationService} from '../dataManipulationService';
+import {DateTimeHelper} from '../../helpers/dateTimeHelper';
+
+const dataManipulationService = new DataManipulationService();
+const dateTimeHelper = new DateTimeHelper();
+
+export class MagsStandardListService {
+  public manipulatedMagsStandardListData(magsStandardListData: object, language: string, languageFile: string): object {
+    let hearingCount = 0;
+    magsStandardListData['courtLists'].forEach(courtList => {
+      courtList['courtHouse']['courtRoom'].forEach(courtRoom => {
+        courtRoom['session'].forEach(session => {
+          session['sittings'].forEach(sitting => {
+            const allHearings = [];
+            session['formattedJudiciaries'] = dataManipulationService.findAndManipulateJudiciary(sitting);
+            this.formatCaseTime(sitting, 'h:mma');
+            hearingCount = hearingCount + sitting['hearing'].length;
+            sitting['hearing'].forEach(hearing => {
+              sitting['formattedDuration'] = dateTimeHelper.formatDuration(sitting['durationAsDays'] as number,
```

Comment:
The method formatDuration() does not need anything from 'hearing'. May be it could be moved to the line before 'sitting['hearing].forEach' so it does not need to recalculated for each hearing.

Reply:
fixed.